classes/session/SessionManager.inc.php

```diff
@@ -53,6 +59,14 @@ function __construct($sessionDao, $request) {
 			array($this, 'gc')
 		);
 
+		// Check if the session is tied to the parent domain
```

Comment:
I would expect the `config.inc.php` override to apply to this as well when it's provided, no?

Reply:
Wait.  How would `$this->userSession` ever be already set in the constructor?  Can `session_set_save_handler()` somehow intervene to call `$this->read()` even before the object is instantiated?  This seems off.

Reply:
@ctgraham I was debugging this and it seems you are correct yesterday and you are right, I could not see any situation where $this->userSession was set and the check would actually work

Reply:
Because as far as I can see, this is the exact code that should take care of subdomains, right?

Reply:
Oh, and if you move that piece of code to it's original place, after session_start, you get:
`PHP Warning: ini_set(): A session is active. You cannot change the session module’s ini settings at this time in lib/pkp/classes/session/SessionManager.inc.php on line 76,`

Reply:
What if we do this in the beginning of construct. Seems to worki

```
		$cookieDomain = $request->getServerHost(null, false);
		$siteBaseUrlHost = parse_url(Config::getVar('general', 'base_url'), PHP_URL_HOST);
		if (strtolower(substr($cookieDomain, -1 - strlen($siteBaseUrlHost))) == '.'.strtolower($siteBaseUrlHost)) {
			$cookieDomain = $siteBaseUrlHost;
		}
		ini_set('session.cookie_domain', $cookieDomain);
```

Reply:
That would be the only required change. No new config setting needed.

Basically we are comparing the base_url setting and the getServerHost. If getServerHost returns a subdomain, the code will compare it with the base_url setting and use the base_url instead. 

If there would be an independent journal domain, getServerHost would return that and would use it like it does with the current code, creating a whole new session.

If the base_url setting is not configured correctly (OJS works even if you leave the pkp.sfu.ca url there) it should cause a problem

Do you see a problem @ctgraham ?

Reply:
See 
https://github.com/pkp/pkp-lib/pull/6315
https://github.com/pkp/ojs/pull/2895